File: RepoTransBench/target_projects/Python/C++/BBuf_onnx_learn/tests/original/test_tools_tool.cpp
#include "tools/tool.h"
#include <gtest/gtest.h>
#include <string>

TEST(TestToolsTool, BasicOps) {
    EXPECT_EQ(tool::add(10, 5), 15);
    EXPECT_EQ(tool::sub(10, 5), 5);
    EXPECT_EQ(tool::mul(4, 0), 0);
    EXPECT_EQ(tool::div(20, 4), 5);
}

TEST(TestToolsTool, NegativeValues) {
    EXPECT_EQ(tool::sub(-5, -5), 0);
    EXPECT_EQ(tool::mul(-2, 3), -6);
}

TEST(TestToolsTool, DivZero) {
    EXPECT_THROW(tool::div(1, 0), std::logic_error);
}

TEST(TestToolsTool, ToolMultiply) {
    tool::Tool t;
    EXPECT_EQ(t.multiply(-1, 8), -8);
}

TEST(TestToolsTool, ToolIdentity) {
    tool::Tool t;
    EXPECT_EQ(t.identity(std::string("abc")), "abc");
}

File: RepoTransBench/target_projects/Python/Rust/BBuf_onnx_learn/tests/original/test_tools_tool.rs
use bbuf_onnx_learn::tools::tool::{add, sub, mul, div, Tool};

#[test]
fn test_basic_ops() {
    assert_eq!(add(10, 5), 15);
    assert_eq!(sub(10, 5), 5);
    assert_eq!(mul(4, 0), 0);
    assert_eq!(div(20, 4), 5);
}

#[test]
fn test_negative_values() {
    assert_eq!(sub(-5, -5), 0);
    assert_eq!(mul(-2, 3), -6);
}

#[test]
fn test_div_zero() {
    let result = std::panic::catch_unwind(|| { div(1, 0); });
    assert!(result.is_err());
}

#[test]
fn test_tool_multiply() {
    let t = Tool;
    assert_eq!(t.multiply(-1, 8), -8);
}

#[test]
fn test_tool_identity() {
    let t = Tool;
    assert_eq!(t.identity(String::from("abc")), String::from("abc"));
}